src/app-state.ts

```diff
@@ -376,29 +384,101 @@ function useCartItemsState() {
 
   const updateCartItems = () => {
     getCartItems(mcart).then(res => {
-      setCartData(res.data.filter(({ type }) => type === 'cart_item' || type === 'custom_item'));
-      setPromotionItems(res.data.filter(({ type }) => type === 'promotion_item'));
-      setCount(res.data.reduce((sum, { quantity }) => sum + quantity, 0));
-      setTotalPrice(res.meta.display_price.without_tax.formatted)
+      const cartData = res.data.length ? res.data.filter(({ type }) => type === 'cart_item' || type === 'custom_item') : [];
+      setCartData(cartData);
+      const promotionItems = res.data.length ? res.data.filter(({ type }) => type === 'promotion_item') : [];
+      setPromotionItems(promotionItems);
+      const count = res.data.length ? res.data.reduce((sum, { quantity }) => sum + quantity, 0) : 0;
+      setCount(count);
+      const totalPrice = res.meta ? res.meta.display_price.without_tax.formatted : '';
+      setTotalPrice(totalPrice)
     });
   };
 
   return { cartData, promotionItems, count, totalPrice, updateCartItems }
 }
 
+function useMultiCartDataState() {
+  const token = localStorage.getItem('mtoken') || '';
+  const mcustomer = localStorage.getItem('mcustomer') || '';
+  const [multiCartData, setMultiCartData] = useState<moltin.CartItem[]>([]);
+  const [selectedCartName, setSelectedCartName] = useState('');
+  const [isCartSelected, setIsCartSelected] = useState(false);
+
+  useEffect(() => {
+    if (token) {
+      getMultiCarts(token).then(res => {
+        setMultiCartData(res.data);
+        const cartName = res.data[0] ? res.data[0].name : '';
+        const cartId = res.data[0] ? res.data[0].id : '';
+        updateSelectedCartName(cartName);
+        localStorage.setItem('mcart', cartId);
+      });
+    }
+    else {
+      clearCartData();
+    }
+  }, [mcustomer, token]);
+
+  const createCart = (data: any) => (
+    createNewCart(data, token).then((cartRes: any) =>
+      addCustomerAssociation(cartRes.data.id, mcustomer, token).then(() =>
```

Comment:
Thank you!

Reply:
We need to ensure when a "guest" shopper logs-in as a registered shopper, we need to associate their cart to their customer ID as well. We can re-use the function you have here for this.

flow example: 

- Add product to cart as guest shopper
- login as registered user
- cart created as guest is now associated to registered shopper and can be accessed in "multi-cart" view

Reply:
@shaunmaharaj It's a great idea. I added this functionality